.github/workflows/pr-checks.yml

```diff
@@ -0,0 +1,19 @@
+name: PR Checks
+
+on: pull_request
+
+jobs:
+  pr-checks:
+    runs-on: macos-latest
```

Comment:
We'll expand this later to include Linux and Windows. Currently the test suite has only been fully checked against MacOS locally.